pytensor/link/pytorch/linker.py

```diff
@@ -13,7 +13,12 @@ def input_filter(self, inp: Any) -> Any:
         return pytorch_typify(inp)
 
     def output_filter(self, var: Variable, out: Any) -> Any:
-        return out.cpu()
+        from torch import is_tensor
+
+        if is_tensor(out) and out.device.type == "cpu":
+            return out.detach().numpy()
+        else:
+            return out
```

Comment:
We should add an option to the PyTorch linker: `coerce_to_numpy` that defaults to `True`, and will apply to either CPU or GPU tensors? When False we don't force any conversion.

This mixed behavior is just unpredictable no?

Reply:
I like it.